frontend/app/(defaultLayout)/components/page-frame/index.tsx

```diff
@@ -0,0 +1,57 @@
+'use client';
+
+import { Box, Flex } from '@chakra-ui/react';
+import { AppProviders } from '@components/common/AppProviders';
+import { SHOPIFY_STOREFRONT_VERSION } from '@config/env';
+import { ShopifyStorefrontProvider } from '@ifixit/shopify-storefront-client';
+import type { DefaultLayoutProps } from '@layouts/default/server';
+import type { ReactNode } from 'react';
+import { AnalyticsScripts } from '../analytics';
+import { Footer } from './footer';
+import { Header } from './header';
+
+interface PageFrameProps {
+   children: ReactNode;
+   regionMenu: React.ReactNode;
+   layoutData: DefaultLayoutProps;
+}
+
+export function PageFrame({
+   children,
+   regionMenu,
+   layoutData,
+}: PageFrameProps) {
+   const {
+      currentStore: { header, footer, socialMediaAccounts },
+      shopifyCredentials,
+      globalSettings,
+   } = layoutData;
+
+   return (
+      <AppProviders>
+         <ShopifyStorefrontProvider
+            shopDomain={shopifyCredentials.storefrontDomain}
+            storefrontAccessToken={shopifyCredentials.storefrontAccessToken}
+            apiVersion={SHOPIFY_STOREFRONT_VERSION}
+         >
+            <Box>
+               <Flex direction="column" minH="100vh">
+                  <Header menu={header.menu} />
+                  {children}
```

Comment:
I notice that we lost the admin message in this refactor. That's ok, but we should make an issue to re-imagine this feature or implement it in the app router later on.

Reply:
Yes, this is part of a longer refactor to move the product page to the app router. I've done much of the work during these holidays, now I'm splitting it up in chunks to make it easier to review and ship